### PR TITLE
Generate a more unique texture name for glb embedded textures

### PR DESCRIPTION
**Pull request by iche033**
# 🦟 Bug fix

## Summary

For glb meshes, our assimp loader generates a unique texture name for each embedded texture by prepending the 'scene' or 'root node' name of the glb mesh. However, the generated name may not always be unique. It's possible that different glb files have the same scene and root name. This PR adds the file base name of the mesh to the generated texture name string to make it more unique.

I think this change is similar to the intention in https://github.com/gazebosim/gz-common/pull/565 where it's also trying to make texture names unique.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: graphics/src/AssimpLoader.cc [hunk 1]
```diff
@@ -69,10 +69,12 @@ class AssimpLoader::Implementation
...
   /// \param[in] _matIdx index of the material in the scene
   /// \param[in] _path path where the mesh is located
+  /// \param[in] _fileBaseName Base name of the mesh file.
   /// \return pointer to the converted common::Material
-  public: MaterialPtr CreateMaterial(const aiScene* _scene,
+  public: MaterialPtr CreateMaterial(const aiScene *_scene,
                                      unsigned _matIdx,
-                                     const std::string& _path) const;
+                                     const std::string &_path,
+                                     const std::string &_fileBaseName) const;
```
Comment (ahcorde) [inline graphics/src/AssimpLoader.cc:77]: This is breaking API in `gz-common5`, you should target `main`, isn't it ?

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: graphics/src/AssimpLoader.cc (hunk 1 above)
Comment (iche033) [reply graphics/src/AssimpLoader.cc:77]: this function is in the private `AssimpLoader::Implementation` class and not exposed in header. So should be fine?

File: graphics/src/AssimpLoader.cc [hunk 2]
```diff
@@ -695,12 +706,18 @@ Mesh *AssimpLoader::Load(const std::string &_filename)
   }
   auto& rootNode = scene->mRootNode;
   auto rootName = ToString(rootNode->mName);
-
-  // compute assimp root node transform
-  std::string extension = _filename.substr(_filename.rfind(".") + 1,
-      _filename.size());
+  auto fileBaseName = common::basename(_filename);
+  std::string extension;
+  std::size_t extIdx = _filename.rfind(".");
```
Comment (ahcorde) [inline graphics/src/AssimpLoader.cc:711]: include `<cstddef>` and `<string>`

**Review by ahcorde — CHANGES_REQUESTED** [review]
just a minor change

File: graphics/src/AssimpLoader.cc (hunk 2 above)
Comment (iche033) [reply graphics/src/AssimpLoader.cc:713]: added. [163222b](https://github.com/gazebosim/gz-common/pull/606/commits/163222b1bf446210069db05f8742220981d2b4ef)

**Review by ahcorde — APPROVED** [review]

**Pull request merged**